### PR TITLE
update utils and examples

### PR DESCRIPTION
**Pull request by saidbleik**
### Description
- Updated NER example to improve usability based on feedback.
- Switched to automodels in transformers to support a wider range of pre-trained models

### Checklist:

- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](https://github.com/microsoft/nlp-recipes/blob/master/CONTRIBUTING.md).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/named_entity_recognition/ner_wikigold_transformer.ipynb [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "\n",
```
Comment (daden-ms) [inline examples/named_entity_recognition/ner_wikigold_transformer.ipynb:28]: <p>It's better to have this description should occur before the definition of DATA_URL. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/nlp-recipes/pull/586/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (daden-ms) [inline examples/named_entity_recognition/ner_wikigold_transformer.ipynb:28]: <p>Maybe add a cell to show an example of sentence list and label list and add this link https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging) to guide users who need to create their own labeled data</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/nlp-recipes/pull/586/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: utils_nlp/models/transformers/common.py [hunk 2]
```diff
@@ -39,6 +52,18 @@
 TOKENIZER_CLASS.update(
```
Comment (daden-ms) [inline utils_nlp/models/transformers/common.py:52]: I tried AutoTokenizer in the newest version of Transformers and found the code is much cleaner without specifically using such a dictionary.

File: utils_nlp/models/transformers/named_entity_recognition.py [hunk 3]
```diff
@@ -77,7 +87,7 @@ def get_inputs(batch, device, model_name, train_mode=True):
                 Labels are only returned when train_mode is True.
         """
         batch = tuple(t.to(device) for t in batch)
-        if model_name.split("-")[0] in ["bert", "distilbert"]:
+        if model_name in list(TC_MODEL_CLASS):
```
Comment (daden-ms) [inline utils_nlp/models/transformers/named_entity_recognition.py:90]: model_name is not optional in this function. The documentation needs to be changed.

File: utils_nlp/models/transformers/named_entity_recognition.py [hunk 4]
```diff
@@ -187,6 +195,10 @@ def _is_iterable_but_not_string(obj):
             )
             max_len = MAX_SEQ_LEN
 
+        logging.warn(
```
Comment (daden-ms) [inline utils_nlp/models/transformers/named_entity_recognition.py:198]: Does this function only work for bert? If it works for other models, the name is misleading.

File: utils_nlp/models/transformers/named_entity_recognition.py [hunk 5]
```diff
@@ -233,11 +245,6 @@ def _is_iterable_but_not_string(obj):
                     new_tokens.append(sub_word)
 
             if len(new_tokens) > max_len:
-                logging.warn(
-                    "Text after tokenization with length {} has been truncated".format(
-                        len(new_tokens)
-                    )
-                )
                 new_tokens = new_tokens[:max_len]
                 new_labels = new_labels[:max_len]
```
Comment (daden-ms) [inline utils_nlp/models/transformers/named_entity_recognition.py:249]: function fit_to_block_size and build_mask from abstractive_summarization_bertsum.py (originally from huggingface's transformers) can be reused here and probably much easier to comprehend.

Comment (daden-ms) [reply utils_nlp/models/transformers/named_entity_recognition.py:249]: I feel processing the dataset per example is much cleaner way to see what preprocessing is needed. Also if you choose to process the dataset one example at a time, there is no need to return a TensorDataset, a list of dictionary suffices, and in your get_inputs function, you can specify the field, instead of using the index 0, 1 to track what the inputs requires. It's up to you to decide whether you want to make this change or not.

File: utils_nlp/models/transformers/named_entity_recognition.py [hunk 6]
```diff
@@ -398,7 +405,9 @@ def fit(
 
         # init scheduler
         scheduler = Transformer.get_default_scheduler(
-            optimizer=self.optimizer, warmup_steps=warmup_steps, num_training_steps=max_steps
+            optimizer=self.optimizer,
```
Comment (daden-ms) [inline utils_nlp/models/transformers/named_entity_recognition.py:408]: should we use dataset instead of dataloader to make it consistent with other transformer models?

Comment (daden-ms) [reply utils_nlp/models/transformers/named_entity_recognition.py:408]: same for prediction

Comment (daden-ms) [reply utils_nlp/models/transformers/named_entity_recognition.py:408]: can the predict function take the label_map to provide the entity labels directly without having the users call the get_predicted_token_labels?

File: utils_nlp/models/transformers/named_entity_recognition.py (hunk 3 above)
Comment (daden-ms) [reply utils_nlp/models/transformers/named_entity_recognition.py:90]: can do_lower_case be handled within the function, instead of being passed as an argument to avoid mismatch?

File: utils_nlp/models/transformers/common.py (hunk 2 above)
Comment (saidbleik) [reply utils_nlp/models/transformers/common.py:52]: Oh I didn't know these exist - great. I'll try this and might also move to AutoModels while we're at it

**Comment by saidbleik** [comment]
> 
> 
> looks good to me. Just one last comment, do we want to check-in the output of the cells in the Jupiter notebooks?

In general, yes. The NER notebook now includes cells' output.
